org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JavaClientConnection.java

```diff
@@ -162,7 +163,12 @@ public void sendStatus(ServiceStatus serverStatus, String status) {
 	 *            The progress report to send back to the client
 	 */
 	public void sendProgressReport(ProgressReport progressReport) {
-		client.sendProgressReport(progressReport);
+		var preferenceManager = JavaLanguageServerPlugin.getPreferencesManager();

```

Comment:
I'm not sure if that is the correct approach to resolve the client capabilities by using `JavaLanguageServerPlugin.getPreferencesManager()` here.

Reply:
I don't think this helper method is used anywhere in our own code. It seems we have access to the `JavaLanguageClient` itself so there's no need for this.

Based on the name, I would think the method should continue to call `client.sendProgressReport(..)` and we should create a separate helper for `notifyProgress(..)` for the sake of consistency. But it's probably more convenient (if ever used in the future) to have it automatically handle the logic for using the old/new API.

Reply:
If the code isn't used at all shouldn't it be removed?

Reply:
I'm fine with keeping it for future uses. Maybe there's places where they don't have easy access to the underlying client.